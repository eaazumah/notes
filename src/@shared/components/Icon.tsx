import React from 'react';
import {Text} from 'react-native-svg';
import {IconType} from '../@types';
import {colors} from '../constants';
import getIconByType from '../helpers/getIconByType';

export interface IconProps {
  name: string;
  size?: number;
  color?: string;
  type?: IconType;
}

const Icon: React.FC<IconProps> = props => {
  const IconByType = getIconByType(props.type);
  return (
    <React.Suspense fallback={<Text />}>
      <IconByType
        testID="icon"
        name={props.name}
        size={props.size}
        color={props.color}
      />
    </React.Suspense>
  );
};

Icon.defaultProps = {
  size: 30,
  type: 'Ionicons',
  color: colors.black,
};

export default React.memo(Icon);
