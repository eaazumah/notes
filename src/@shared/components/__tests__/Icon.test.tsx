import {render} from '@testing-library/react-native';
import React from 'react';
import Icon, {IconProps} from '../Icon';

function setup(props: IconProps) {
  return render(<Icon {...props} />);
}

test('should render icon', async () => {
  const screen = setup({
    size: 35,
    name: 'add',
    type: 'Ionicons',
  });

  expect(screen.toJSON()).toMatchSnapshot();
});
