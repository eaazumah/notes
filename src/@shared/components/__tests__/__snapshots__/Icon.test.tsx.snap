// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render icon 1`] = `
<RNSVGText
  content={null}
  dx={Array []}
  dy={Array []}
  font={Object {}}
  rotate={Array []}
  x={Array []}
  y={Array []}
/>
`;
